packages/yarnpkg-core/sources/scriptUtils.ts

```diff
@@ -38,7 +38,7 @@ async function makePathWrapper(location: PortablePath, name: Filename, argv0: Na
   await xfs.chmodPromise(ppath.join(location, name), 0o755);
 }
 
-async function detectPackageManager(location: PortablePath) {
+async function detectPackageManager(location: PortablePath, stdout: any, projectName: string) {
```

Comment:
I'd prefer `detectPackageManager` to stay without side effects. Can you rather return an object?

```ts
interface PackageManagerSelection {
  packageManager: PackageManager;
  reason: string;
}
```

Reply:
Also, no need to duplicate the project name, which you already added below.

Reply:
Agreed, updated.